### DISCUSSION TITLE
Question about licensing

### DISCUSSION THREAD

**Discussion by realtebo** [Laravel-Backpack/community-forum#549] [Q&A (Help)] [answered]
We bought a v4 and a v5 in the past years licenses for unlimite projects.
At today, our license is exprired
We were waiting v6 to bought a new license.
Question: while license will be active again, do we receive updates for v4 and v5 also?

**Comment by pxpm** [comment]
Hey @realtebo thanks for the question.

Our licence is: "access to code you bought forever, access to updates for one year and keep them". 

It means that when you buy the license you get access to the previous code, and the code we will push for the next year. 

In your example, if your license expired today, you have access to all the code we pushed until yesterday, forever, without renewing the license. The caveat is that you don't receive further updates. 

If you renew the license you will keep getting updates for your products for one year (in your case, all products since you have unlimited).

I think the system is designed to give you access to previous versions (v4 and v5) if you have access to v6. But @tabacitu can better confirm that. 

Cheers

**Reply by tabacitu** [reply, marked as answer]
> you have access to all the code we pushed until yesterday, forever, without renewing the license. The caveat is that you don't receive further updates.

Condirmed! 100%. Well phrased Pedro.

> the system is designed to give you access to previous versions (v4 and v5) if you have access to v6

Yes it is. But @realtebo you have access to v4 and v5 thanks to your previous purchases. Any updates in theory but in practice we DO NOT plan to push any more updates to v4 or v5. Starting this v6 launch:
- v6 receives both bug fixes and security fixes;
- v5 only receives security fixes;
- V4 won’t receive any fixes;

You can see that in Backpack/CRUD in SECURITY_POLICY.MD but I’m on mobile now, can’t link, sorry.

Hope this clears things up.

**Reply by realtebo** [reply]
Thanks to both. We will buy new license in 2-3 days.

We have A LOT of v5 out there, and actually we have fixed max version to the latest minor we have access, because we already faced an impossible to install situation while deploying a new instance of a webapp to a new customer. So we noticed our license of v5 was, as expetected, stopped to give use new minors, but actually we are already missine a few minor releases of v5 and we'd like to get all of them, specially if they are security fixes.

so, it's a good news !,  thanks
